src/mon/OSDMonitor.cc

```diff
@@ -7829,14 +7829,30 @@ int OSDMonitor::prepare_new_pool(string& name,
 				 const string& pg_autoscale_mode,
 				 ostream *ss)
 {
+  const string default_autoscale_mode = g_conf().get_val<string>("osd_pool_default_pg_autoscale_mode");
   if (name.length() == 0)
     return -EINVAL;
-  if (pg_num == 0)
-    pg_num = g_conf().get_val<uint64_t>("osd_pool_default_pg_num");
+  if (pg_num == 0) {
+    if (pg_autoscale_mode.empty()) {
+      if (default_autoscale_mode == "on") {
+         pg_num = 1;
+      }
+      else {
+        pg_num = g_conf().get_val<uint64_t>("osd_pool_default_pg_num");
+      }
+    }
+    else if (pg_autoscale_mode == "on") {
+      pg_num = 1;
+    }
+    else {
+      pg_num = g_conf().get_val<uint64_t>("osd_pool_default_pg_num");
+    }
+  }
```

Comment:
nit, could also revise this block so it is more compact, like
```c++
  if (pg_num == 0) {
    auto pg_num_from_mode =
      [pg_num=g_conf().get_val<uint64_t>("osd_pool_default_pg_num")]
      (const string& mode) {
      return mode == "on" ? 1 : pg_num;
    };
    pg_num = pg_num_from_mode(
      pg_autoscale_mode.empty() ?
      g_conf().get_val<string>("osd_pool_default_pg_autoscale_mode") :
      pg_autoscale_mode);
  }
```

Reply:
This is really nice and compact